### ISSUE TITLE
invalid redirect_uri. This value must match a URL registered with the API Key.

### ISSUE THREAD

**Issue by laren2j** [sahat/satellizer#437]
I am integrating linkedin login, and i am getting this error. I try this with the usual method and seems to be working fine. Can anyone tell me where i am going wrong?

OAuth 2.0 : Authorized Redirect URLs: 
http://localhost/

and in my controller i am triggering this on click
this is my button in view
<button class="btn btn-lg btn-danger btn-block" ng-click="misc.authenticate('linkedin')">Linkedin</button>

and on click 

    $scope.authenticate = function(provider) {
        $auth.authenticate(provider);
    };

I get the window for linked in permission but with an error : invalid redirect_uri. This value must match a URL registered with the API Key.

I have checked my settings on linked in my apps

![linkedinpopup](https://cloud.githubusercontent.com/assets/1692096/7880398/a7e40994-0616-11e5-9ed8-154467cacbc6.png)
.

**Comment by laren2j** [comment]
i figured the issue but need help on how to make the necessary changes in library changes.

I get this because the redirect_uri is 

&redirect_uri=http://localhost&state=STATE     (in the url of the pop up window)

when i change this to

&redirect_uri=http://localhost/&state=STATE    (if you observe there is a ' / ' after localhost)

**Comment by sahat** [comment]
@laren2j there is no need to change anything in the library. Have you tried updating OAuth 2.0 Authorized Redirect URLs to `http://localhost` without the slash?

You can always override `redirectUri` too. The default is:
```
redirectUri: window.location.origin || window.location.protocol + '//' + window.location.host,
```

So, in other words for you it would be:

```
redirectUri: window.location.origin + '/' || window.location.protocol + '//' + window.location.host + '/',
```

**Comment by sahat** [comment]
I have outlined all possible options as well as their default values in the README:

https://github.com/sahat/satellizer#configuration

**Comment by mixn** [comment]
Hi @sahat,

Thank you for this great tool, it’s very convenient. :)

Any idea why I always get redirected to the root, no matter what?

Inside my `app.js`’s `config` block I have this:

```
$authProvider.facebook({
  clientId: 'myclientIDhere',
  redirectUri: window.location.origin + '/' || window.location.protocol + '//' + window.location.host + '/'
});
```

and inside `partner.view.html` this:
```
<div ng-controller="PartnerCtrl as PC">
  <button ng-click="PC.login('facebook')">Log in with Facebook</button>
</div>
```

I have tried most options I’ve found online already, most of them we’re way on how to prevent the default behavior of an `a` tag. Then I switched to a `button` as seen in your live demo and, after the server has successfully responded, I still get redirected to the root.

Any advice here? Would really appreciate it, thank you. :+1:
